tests/virt_autotest/kubevirt_tests_server.pm

```diff
@@ -160,34 +181,49 @@ sub check_service_status {
     assert_script_run("! journalctl -u rke2-server | grep \'\"level\":\"error\"\'");
 }
 
-sub install_kubevirt_pkgs_n_deploy_manifests {
+sub install_kubevirt_packages {
     my $self = shift;
-    # Install and deploy required kubevirt packages and manifests
-    my $os_version = get_required_var('VERSION');
+    # Install required kubevirt packages
+    my $os_version = get_var('VERSION');
     my $virt_tests_repo = get_required_var('VIRT_TESTS_REPO');
     my $virt_manifests_repo = get_var('VIRT_MANIFESTS_REPO');
 
-    record_info('Install kubevirt packages and deploy manifests', '');
+    record_info('Install kubevirt packages', '');
     # Development Tools repo for OBS Module, e.g. http://download.suse.de/download/ibs/SUSE/Products/SLE-Module-Development-Tools-OBS/15-SP4/x86_64/product/
     # Development product test repo for SLE official product OSD testing, e.g. http://download.suse.de/ibs/SUSE:/SLE-15-SP4:/GA/standard/
     # Devel test repo, e.g. http://download.suse.de/download/ibs/Devel:/Virt:/SLE-15-SP4/SUSE_SLE-15-SP4_Update_standard/
     # MU product test (SLE official MU channel+incidents)
+    transactional::enter_trup_shell(global_options => '--drop-if-no-change') if (is_transactional);
     zypper_call("ar $virt_tests_repo Virt-Tests-Repo");
     zypper_call("ar $virt_manifests_repo Virt-Manifests-Repo") if ($virt_manifests_repo);
     zypper_call("--gpg-auto-import-keys ref");
 
     my $virt_manifests = 'containerized-data-importer-manifests kubevirt-manifests kubevirt-virtctl';
+    my $search_manifests = $virt_manifests =~ s/\s+/\\\|/gr;
 
     if ($virt_manifests_repo) {
         zypper_call("in -f -r Virt-Manifests-Repo $virt_manifests");
     } elsif (script_run("rpmquery $virt_manifests")) {
-        zypper_call("in -f -r SLE-Module-Containers${os_version}-Updates $virt_manifests");
+        if (is_transactional || script_run("zypper se -r SLE-Module-Containers${os_version}-Updates $virt_manifests | grep -w '$search_manifests'")) {
```

Comment:
Is the if condition given correct?  I guess it should be 
```if (is_transactional || ! script_run("zypper se -r SLE-Module-Containers${os_version}-Updates $virt_manifests | grep -w '$search_manifests'"))```

Reply:
Yes, it was verified. You can see it will return '0' if the repo included those packages.
Refer to http://10.67.129.66/tests/1735#step/kubevirt_tests_server/84

vh013:~ # zypper se -r SLE-Module-Containers15-SP5-Updates containerized-data-importer-manifests kubevirt-manifests kubevirt-virtctl |grep -w 'containerized-data-importer-manifests\|kubevirt-manifests\|kubevirt-virtctl'
i+ | containerized-data-importer-manifests | YAML manifests used to install CDI      | package
i+ | kubevirt-manifests                    | YAML manifests used to install kubevirt | package
i+ | kubevirt-virtctl                      | Client for managing kubevirt            | package
vh013:~ # echo $?
0

Reply:
Yes, it was verified. You can see it will return '0' if the repo included those packages.
Refer to http://10.67.129.66/tests/1735#step/kubevirt_tests_server/84

Reply:
Yes, it was verified. You can see it will return '0' if the repo included those packages.
Refer to http://10.67.129.66/tests/1735#step/kubevirt_tests_server/84

vh013:~ # zypper se -r SLE-Module-Containers15-SP5-Updates containerized-data-importer-manifests kubevirt-manifests kubevirt-virtctl |grep -w 'containerized-data-importer-manifests\|kubevirt-manifests\|kubevirt-virtctl'
i+ | containerized-data-importer-manifests | YAML manifests used to install CDI      | package
i+ | kubevirt-manifests                    | YAML manifests used to install kubevirt | package
i+ | kubevirt-virtctl                      | Client for managing kubevirt            | package
vh013:~ # echo $?
0

Reply:
Yep, sorry, my bad.

Reply:
That's ok. NP